src/3D/ft3dio.c

```diff
@@ -42,27 +42,30 @@ static int      firstFID = TRUE;
 extern char	*userdir;
 extern int	maxfn12;	/* maximum F1-F2 real Fourier number	*/
 extern int      mapFIDblock(int block_no);
+extern int removelock(char *filepath);
+extern int createlock(char *filepath, int type);
 
 /*---------------------------------------
 |                                       |
 |           readFIDdata()/12            |
 |                                       |
 +--------------------------------------*/
-int readFIDdata(fd, data, wspace, filehead, nfidbytes, npadj, lsfval,
-                        lastfid, fid_nbr, dpflag, fidmap, pf3acq, lpval)
-char            *wspace;	/* pointer to workspace memory		*/
-int             fd,		/* FID file descriptor			*/
-		pf3acq,		/* Acq-F3proc flag			*/
-                nfidbytes,	/* bytes per t3 FID			*/
-                *lastfid,	/* last FID currently processed		*/
-		fid_nbr,	/* which FID to get (first=0) */
-                npadj,		/* number of adjusted complex td points	*/
-                lsfval,		/* complex points to be left-shifted	*/
-                dpflag,		/* double-precision flag for FID data	*/
-		fidmap;		/* flag for using FID map		*/
-float           *data;		/* pointer to converted FID data	*/
-dfilehead	*filehead;	/* pointer to FID file header		*/
-float		*lpval;		/* lpval from FID block header (DSP)	*/
+int readFIDdata(int fd, float *data, char *wspace, dfilehead *filehead,
```

Comment:
I came across this note about 'float' vs. 'double' in the k&r C ... (http://ee.hawaii.edu/~tep/EE160/Book/chapapx/node7.html) not sure if it's relevant here, I saw somewhere (can't find it anymore) that declaring float in k&r-style function declarations would actually declare a double.  Made a little test that appears, at least in clang, not to resize float, regardless of function declaration style.  This isn't a complaint, just a note to "double"-check on gcc (on the laptop now.. only clang).

#include <stdio.h>

f1(v)
float v;
{
  printf("sizeof(v) = %zd\n", sizeof(v));
}

void f2(float v)
{
  printf("sizeof(v) = %zd\n", sizeof(v));
}

int main(int argc, char *argv[])
{
  f1(0);
  f2(0);
}

Reply:
I ran your test program with gcc version 4.4.7 (CentOS 6), gcc 4.8.5 (CentOS 7.4) and gcc 7.2.0 (Ubuntu 17) and it worked correctly in each case.